File: from-books/recursion/language/javascript/exercise-2.ts
function printNumbers(data: any): void {
  if (Array.isArray(data)) {
    data.forEach((item) => printNumbers(item));
  } else {
    console.log(data);
  }
}

function main(): void {
  const mixed = [1, 2, 3, [4, 5, 6], 7, 8, [9, [10, 11]]];

  printNumbers(mixed);
}

main();

File: from-books/recursion/language/python/exercise-2.py
def print_numbers(data):
    # Check if data is a list (or any iterable)
    if isinstance(data, list):
        for item in data:
            print_numbers(item)
    else:
        print(data)


def main():
    mixed = [
        1,
        2,
        3,
        [4, 5, 6],
        7,
        8,
        [
            9,
            [10, 11],
        ],
    ]

    print_numbers(mixed)


if __name__ == "__main__":
    main()
